docs/ref-manual/cvl/invariants.md

```diff
@@ -28,29 +35,269 @@ preserved_block ::= "preserved"
 Overview
 --------
 
-```{todo}
-This section is incomplete.  See [the user guide](/docs/user-guide/bank/index)
-for an overview of invariants.
+In CVL, an invariant is a property of the contract state that is expected to be
+true whenever a contract method is not currently executing.  This kind of
+invariant is sometimes called a "representation invariant".
+
+Each invariant has a name, possibly followed by a set of parameters, followed
+by a boolean expression.  We say the invariant *holds* if the expression
+evaluates to true in every reachable state of the contract, and for all
```

Comment:
the word "reachable" is a bit misleading here because the tool does not know what are the reachable states, so it checks for all states. It's up to the user to refine that set of states. I agree it's a bit of a fine point but maybe worth using a precise language here.